.github/workflows/integration-tests.yml

```diff
@@ -213,11 +226,24 @@ jobs:
           npm install $(echo $(cat package.json | jq -r --arg version $CANARY_VERSION '["@grafana/eslint-config", "@grafana/tsconfig", "@grafana/scenes", "@grafana/experimental"] as $blacklist | [(.devDependencies,.dependencies) | keys] | flatten | unique | map(select ( test("^@grafana") ) ) as $deps | $deps - $blacklist | map("\(.)@\($version)") | join(" ")') )
         working-directory: ${{ matrix.pluginDir }}
 
+      - name: Run frontend tests
+        run: |
+          npm run test:ci
+        working-directory: ${{ matrix.pluginDir }}
+
       - name: Build plugin with canary_version @grafana dependencies
         run: |
           npm run build
         working-directory: ${{ matrix.pluginDir }}
 
+      - name: Test plugin backend
+        uses: magefile/mage-action@v2
+        with:
+          version: latest
+          args: -v test
+          workdir: ${{ matrix.pluginDir }}
+        if: steps.backend-check.outputs.MAGEFILE_EXISTS == 'true'
+
```

Comment:
I'm not sure we need to run backend tests twice unless we're also bumping the go dependencies (which I don't think we're doing). 🤔